fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeParquetHandler.java

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.delta;
+
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.Lists;
+import com.starrocks.common.Pair;
+import io.delta.kernel.data.ColumnarBatch;
+import io.delta.kernel.defaults.engine.DefaultParquetHandler;
+import io.delta.kernel.exceptions.KernelEngineException;
+import io.delta.kernel.expressions.Predicate;
+import io.delta.kernel.internal.replay.LogReplay;
+import io.delta.kernel.internal.util.Utils;
+import io.delta.kernel.types.StructType;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.FileStatus;
+import org.apache.hadoop.conf.Configuration;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import static java.lang.String.format;
+
+public class DeltaLakeParquetHandler extends DefaultParquetHandler {
+    private final Configuration hadoopConf;
+    private final LoadingCache<Pair<String, StructType>, List<ColumnarBatch>> checkpointCache;
+
+    public DeltaLakeParquetHandler(Configuration hadoopConf, LoadingCache<Pair<String, StructType>,
+            List<ColumnarBatch>> checkpointCache) {
+        super(hadoopConf);
+        this.hadoopConf = hadoopConf;
+        this.checkpointCache = checkpointCache;
+    }
+
+    public static List<ColumnarBatch> readParquetFile(String filePath, StructType physicalSchema, Configuration hadoopConf) {
+        io.delta.kernel.defaults.internal.parquet.ParquetFileReader batchReader =
+                new io.delta.kernel.defaults.internal.parquet.ParquetFileReader(hadoopConf);
+        CloseableIterator<ColumnarBatch> currentFileReader = batchReader.read(filePath, physicalSchema, Optional.empty());
+
+        List<ColumnarBatch> result = Lists.newArrayList();
+        while (currentFileReader != null && currentFileReader.hasNext()) {
+            result.add(currentFileReader.next());
+        }
+        Utils.closeCloseables(currentFileReader);
+        return result;
+    }
+
+    @Override
+    public CloseableIterator<ColumnarBatch> readParquetFiles(
+            CloseableIterator<FileStatus> fileIter,
+            StructType physicalSchema,
+            Optional<Predicate> predicate) {
```

Comment:
why we don't use predicate

Reply:
we cannot cache the entire parquet data when use predicate

Reply:
is it the same iceberg cache all manifest file content?